server/routes/api/challenges.js

```diff
@@ -0,0 +1,43 @@
+const router = require('express').Router();
+const { Challenge } = require('../../db');
+
+/**  /api/challenges **/
+
+// get all challenges
+router.get('/', (req, res, next) => {
+  Challenge.findAll()
+    .then(challenges => res.send(challenges))
+    .catch(next);
+});
+
+// create a single challenge
+router.post('/', (req, res, next) => {
+  Challenge.create(req.body)
+    .then(challenge => res.send(challenge))
+    .catch(next);
+});
+
+// get a single challenge
+router.get('/:id', (req, res, next) => {
+  Challenge.findByPk(req.params.id)
+    .then(challenge => res.send(challenge))
+    .catch(next);
+});
+
+// update a single challenge
+router.put('/:id', (req, res, next) => {
+  Challenge.findByPk(req.params.id)
```

Comment:
What if id is `NaN`?

Reply:
Doesn't Sequelize coerce to number in this situation? http://docs.sequelizejs.com/class/lib/model.js~Model.html

Reply:
`curl localhost:3000/api/challenges/am_i_coercable?`

Reply:
Its not about an error, its about returning the appropriate response. If you dont receive a number, the database should never hear about it.